test/runtests.jl

```diff
@@ -23,29 +23,44 @@ Random.seed!(100)
     μ ./= sum(μ)
     ν ./= sum(ν)
 
-    # create random cost matrix
-    C = pairwise(SqEuclidean(), rand(1, M), rand(1, N); dims=2)
-
-    # compute optimal transport map and cost with POT
-    pot_P = POT.emd(μ, ν, C)
-    pot_cost = POT.emd2(μ, ν, C)
-
-    # compute optimal transport map and cost with Tulip
-    lp = Tulip.Optimizer()
-    P = emd(μ, ν, C, lp)
-    @test size(C) == size(P)
-    @test MOI.get(lp, MOI.TerminationStatus()) == MOI.OPTIMAL
-    @test maximum(abs, P .- pot_P) < 1e-2
-
-    lp = Tulip.Optimizer()
-    cost = emd2(μ, ν, C, lp)
-    @test dot(C, P) ≈ cost atol = 1e-5
-    @test MOI.get(lp, MOI.TerminationStatus()) == MOI.OPTIMAL
-    @test cost ≈ pot_cost atol = 1e-5
-
-    # ensure that provided map is used
-    cost2 = emd2(similar(μ), similar(ν), C, lp; plan=P)
-    @test cost2 ≈ cost
+    @testset "example" begin
+        # create random cost matrix
+        C = pairwise(SqEuclidean(), rand(1, M), rand(1, N); dims=2)
+
+        # compute optimal transport map and cost with POT
+        pot_P = POT.emd(μ, ν, C)
+        pot_cost = POT.emd2(μ, ν, C)
+
+        # compute optimal transport map and cost with Tulip
+        lp = Tulip.Optimizer()
+        P = emd(μ, ν, C, lp)
+        @test size(C) == size(P)
+        @test MOI.get(lp, MOI.TerminationStatus()) == MOI.OPTIMAL
+        @test maximum(abs, P .- pot_P) < 1e-2
+
+        lp = Tulip.Optimizer()
+        cost = emd2(μ, ν, C, lp)
+        @test dot(C, P) ≈ cost atol = 1e-5
+        @test MOI.get(lp, MOI.TerminationStatus()) == MOI.OPTIMAL
+        @test cost ≈ pot_cost atol = 1e-5
+    end
+
+    @testset "pre-computed plan" begin
+        # create random cost matrix
+        C = pairwise(SqEuclidean(), rand(1, M), rand(1, N); dims=2)
+
+        cost = emd2(μ, ν, C, Tulip.Optimizer())
+
+        # ensure that provided map is used
+        cost2 = emd2(similar(μ), similar(ν), C, lp; plan=P)
```

Comment:
Since the testset changed, I guess the plan `P ` is not defined here. Hence, the test is failling.

Reply:
Yeah, I wanted to restructure the tests a bit but apparently I messed up :smile: `lp` is not defined either...